src/rabbit_reader.erl

```diff
@@ -767,6 +767,7 @@ wait_for_channel_termination(N, TimerRef,
                                "error while terminating:~n~p~n",
                         [self(), ConnName, VHost, User#user.username,
                          CS, Channel, Reason]),
+                    rabbit_core_metrics:channel_closed(ChPid),
```

Comment:
A test in `rabbitmq-management` would be great if we can find a way to make a channel terminate this way.

I suggest we finish this, merge it, and deploy into a long-running environment until we can figure out a way to do this.